Source/WebCore/bindings/js/JSDOMConvertStrings.h

```diff
@@ -170,6 +176,195 @@ template<typename T> struct JSConverter<IDLLegacyNullToEmptyStringAdaptor<T>> {
+};
+
+template<typename T> struct JSConverter<IDLStringContextTrustedScriptAdaptor<T>> {
+    static constexpr bool needsState = true;
+    static constexpr bool needsGlobalObject = false;
+
+    static JSC::JSValue convert(JSC::JSGlobalObject& lexicalGlobalObject, const String& value)
+    {
+        return JSConverter<T>::convert(lexicalGlobalObject, value);
+    }
+};
+
+template<typename T> struct Converter<IDLLegacyNullToEmptyStringStringContextTrustedScriptAdaptor<T>> : DefaultConverter<IDLLegacyNullToEmptyStringStringContextTrustedScriptAdaptor<T>> {
+    static String convert(JSC::JSGlobalObject& lexicalGlobalObject, JSC::JSValue value)
+    {
+        auto result = valueToTrustedScriptString(lexicalGlobalObject, value, true);
+
+        return result.isNull()
+            ? Converter<T>::convert(lexicalGlobalObject, value)
+            : result;
+    }
+};
+
+template<typename T> struct JSConverter<IDLLegacyNullToEmptyStringStringContextTrustedScriptAdaptor<T>> {
+    static constexpr bool needsState = true;
+    static constexpr bool needsGlobalObject = false;
+
+    static JSC::JSValue convert(JSC::JSGlobalObject& lexicalGlobalObject, const String& value)
+    {
+        return JSConverter<T>::convert(lexicalGlobalObject, value);
+    }
+};
+
+template<typename T> struct Converter<IDLStringContextTrustedScriptURLAdaptor<T>> : DefaultConverter<IDLStringContextTrustedScriptURLAdaptor<T>> {
+    static String convert(JSC::JSGlobalObject& lexicalGlobalObject, JSC::JSValue value)
+    {
+        auto result = valueToTrustedScriptURLString(lexicalGlobalObject, value, false);
+
+        return result.isNull()
+            ? Converter<T>::convert(lexicalGlobalObject, value)
+            : result;
+    }
+};
+
+template<typename T> struct JSConverter<IDLStringContextTrustedScriptURLAdaptor<T>> {
+    static constexpr bool needsState = true;
+    static constexpr bool needsGlobalObject = false;
+
+    static JSC::JSValue convert(JSC::JSGlobalObject& lexicalGlobalObject, const String& value)
+    {
+        return JSConverter<T>::convert(lexicalGlobalObject, value);
+    }
+};
+
+template<typename T> struct Converter<IDLAtomStringStringContextTrustedHTMLAdaptor<T>> : DefaultConverter<IDLAtomStringStringContextTrustedHTMLAdaptor<T>> {
+    static AtomString convert(JSC::JSGlobalObject& lexicalGlobalObject, JSC::JSValue value)
+    {
+        auto result = valueToTrustedHTMLAtomString(lexicalGlobalObject, value, false);
+
+        return result.isNull()
+            ? Converter<IDLAtomStringAdaptor<T>>::convert(lexicalGlobalObject, value)
+            : result;
+    }
+};
+
+template<typename T> struct JSConverter<IDLAtomStringStringContextTrustedHTMLAdaptor<T>> {
+    static constexpr bool needsState = true;
+    static constexpr bool needsGlobalObject = false;
+
+    static JSC::JSValue convert(JSC::JSGlobalObject& lexicalGlobalObject, const AtomString& value)
+    {
+        return JSConverter<T>::convert(lexicalGlobalObject, value);
+    }
```

Comment:
TODO: Add the below and update the IDL test to include a reflected URL Trusted type attribute to check it compiles properly.
```
    static JSC::JSValue convert(JSC::JSGlobalObject& lexicalGlobalObject, const URL& value)
    {
        return JSConverter<T>::convert(lexicalGlobalObject, value.string());
    }
    ```